### PR TITLE
Fix: Add customer_address_id to billing_address

### PR DESCRIPTION
**Pull request by UncleTioma**
### Description (*)

`customer_address_id` will not set for new `billing_address`

### Fixed Issues (if relevant)

1. magento/magento2#17705: CustomerAddressId will not set for new billing adress

### Manual testing scenarios (*)

1. Create a new Customer Account
2. Add one Product to Cart
3. Create a New Shipment Address
4. Use Shipment Address as Order Address
5. Check `customer_address_id` with `address_type=billing` in `sales_order_address` 

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=24020) <br/>All committers have signed the CLA.

**Comment by miguelbalparda** [comment]
Hi @UncleTioma! Can you check the failing tests please?

**Comment by UncleTioma** [comment]
> Hi @UncleTioma! Can you check the failing tests please?

Hi @miguelbalparda 
I'm trying to re-run tests, but they cannot finish.
Can you help me with that?

**Comment by UncleTioma** [comment]
@magento run all tests

**Comment by Stepa4man** [comment]
@magento run all tests

**Review by miguelbalparda — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @miguelbalparda, thank you for the review.
[ENGCOM-5741](https://jira.corp.magento.com/browse/ENGCOM-5741) has been created to process this Pull Request
:eight_spoked_asterisk: @miguelbalparda, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Comment by slavvka** [comment]
@magento run all tests

**Comment by slavvka** [comment]
@engcom-Foxtrot please take a look at static and functional EE tests

**Comment by slavvka** [comment]
@engcom-Foxtrot and please cover the change with automated test

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @UncleTioma. Thanks for collaboration. Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. Please cover your fix by automated tests.

**Comment by engcom-Echo** [comment]
I will take care of test coverage

**Comment by engcom-Echo** [comment]
Hi @miguelbalparda, all changes covered by unit tests, could you please review

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5741](https://jira.corp.magento.com/browse/ENGCOM-5741) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Comment by slavvka** [comment]
Need to approve PAT degradation: https://bamboo.corp.magento.com/browse/MCCE24-MCCE24162-1
`Save Order (Admin Create Order): 9.1% (mainline = 11 | team = 12) -> DEGRADATION`
```
SQLs were added to team: 
       UPDATE `quote_address` SET `customer_address_id` = ?, `prefix` = ?, `firstname` = ?, `middlename` = ?, `lastname` = ?, `suffix` = ?, `company` = ?, `street` = ?, `city` = ?, `region_id` = ?, `postcode` = ?, `telephone` = ?, `fax` = ?, `subtotal` = ?, `base_subtotal` = ?, `subtotal_with_discount` = ?, `base_subtotal_with_discount` = ?, `tax_amount` = ?, `base_tax_amount` = ?, `shipping_amount` = ?, `base_shipping_amount` = ?, `shipping_tax_amount` = ?, `base_shipping_tax_amount` = ?, `discount_amount` = ?, `base_discount_amount` = ?, `grand_total` = ?, `base_grand_total` = ?, `shipping_discount_amount` = ?, `base_shipping_discount_amount` = ?, `subtotal_incl_tax` = ?, `discount_tax_compensation_amount` = ?, `base_discount_tax_compensation_amount` = ?, `shipping_discount_tax_compensation_amount` = ?, `shipping_incl_tax` = ?, `base_shipping_incl_tax` = ?, `base_customer_balance_amount` = ?, `customer_balance_amount` = ?, `gift_cards_amount` = ?, `base_gift_cards_amount` = ?, `reward_points_balance` = ?, `base_reward_currency_amount` = ?, `reward_currency_amount` = ? WHERE (address_id=183)
```

**Comment by slavvka** [comment]
PAT failure ticket is created https://jira.corp.magento.com/browse/MC-31650

File: app/code/Magento/Quote/Model/QuoteManagement.php [hunk 1]
```diff
@@ -665,6 +665,10 @@ protected function _prepareCustomerQuote($quote)
         if ($shipping && !$shipping->getCustomerId() && !$hasDefaultBilling) {
             $shipping->setIsDefaultBilling(true);
         }
+
+        if (!$billing->getCustomerAddressId() && $shipping) {
```
Comment (naydav) [inline app/code/Magento/Quote/Model/QuoteManagement.php]: Could you change sequence of the condition?
`($shipping && !$billing->getCustomerAddressId())`

Thanks

**Review by lenaorobei — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5741](https://jira.corp.magento.com/browse/ENGCOM-5741) has been created to process this Pull Request

**Comment by engcom-Foxtrot** [comment]
I've investigated the PAT build degradation. Seems like Billing Address is saved twice in \Magento\Sales\Controller\Adminhtml\Order\Create\Save. On 2.4-dev Billing Address object is not modified between \Magento\Framework\Model\ResourceModel\Db\AbstractDb::save calls, so the second transaction is skipped. While on PR branch Adress's hasDataChanges flag is set to true, so we have one more DB transaction. 
Also, if we are creating Billing Address through admin - no customer_address_id is set to Shipping Address (if it is copy of Billing Address).

**Comment by naydav** [comment]
Hello everyone,
Looks like we need to close the initial issue.
I describe my concern at https://github.com/magento/magento2/issues/17705#issuecomment-596632841
Thanks

**Comment by sidolov** [comment]
@UncleTioma I'm closing the PR according to the @naydav comment. We should refactor code to avoid usage of `customer_address_id ` field instead of adding it to the address. Feel free to open new PR with refactoring if you want to continue.

**Pull request closed without merging**